### PR TITLE
chore: extract sdk server info

### PR DESCRIPTION
**Pull request by jy4096**
fix: https://github.com/numaproj/numaflow/issues/1450

### CODE REVIEW

File: pkg/sdkclient/sinker/client.go [hunk 1]
```diff
@@ -38,23 +36,13 @@ type client struct {
 
 var _ Client = (*client)(nil)
 
-// New creates a new client object.
+// New creates a new client object. Sinker doesn't require server info to start ATM.
```
Comment (whynowy) [inline pkg/sdkclient/sinker/client.go:39]: Is it true? https://github.com/numaproj/numaflow-go/blob/main/pkg/sinker/server.go#L39

File: pkg/sdkclient/source/client/client.go [hunk 2]
```diff
@@ -39,24 +37,14 @@ type client struct {
 
 var _ Client = (*client)(nil)
 
-// New creates a new client object.
+// New creates a new client object. Source doesn't require server info to start ATM.
```
Comment (whynowy) [inline pkg/sdkclient/source/client/client.go:40]: https://github.com/numaproj/numaflow-go/blob/main/pkg/sourcer/server.go#L41

**Review by whynowy — COMMENTED** [review]
`source` and `sourceTransformer` use the same server info path, this needs to be fixed.

File: pkg/sdkclient/sinker/client.go (hunk 1 above)
Comment (jy4096) [reply pkg/sdkclient/sinker/client.go]: The source and sinker clients don't require serverinfo to start.. fixed the godoc

**Review by whynowy — APPROVED** [review]

**Pull request merged**